packages/itwinui-css/src/select/select.scss

```diff
@@ -102,11 +102,16 @@
   align-items: center;
   block-size: 80%;
   max-block-size: calc(var(--iui-size-s) * 3);
+
+  // Truncate when only one tag is present
+  &:only-child {
+    max-inline-size: 100%;
+  }
 }
 
 @mixin iui-select-tag-label {
   @include tag.iui-tag-label;
-  display: inline-flex;
+  display: block;
```

Comment:
i think flexbox was needed for centering. do you want to try `display: flex`?

Reply:
I had changed `inline-flex` to `block` to get the ellipses to show up. Because with `inline-flex` or `flex`, the ellipses didn't show up:

|`block`|`flex`/`inline-flex`|
|-|-|
|![image](https://github.com/iTwin/iTwinUI/assets/45748283/fee87f79-37f0-4374-a5df-083b59d1d98d)|![image](https://github.com/iTwin/iTwinUI/assets/45748283/172d63b7-a8c9-4358-9df4-4248377be051)|

Reply:
flexbox children will refuse to collapse beyond their "real" width by default. could you try adding `min-inline-size: 0` to the child element?

Reply:
As discussed, `select-tag-label` does not have any child tag.

![image](https://github.com/iTwin/iTwinUI/assets/45748283/1e02f16a-6dd4-42fa-a42e-753a0c9d3138)

So, I went ahead with `display: block` instead of `display: inline-flex`, and so also removed the previous `align-items` attribute on `select-tag-label`. (https://github.com/iTwin/iTwinUI/pull/1716/commits/e1956468dbee883b57c17a253f2dbe6ee66a94db)